pkg/kubelet/cm/internal_container_lifecycle_test.go

```diff
@@ -0,0 +1,121 @@
+package cm
+
+import (
+	"github.com/golang/mock/gomock"
+	"k8s.io/api/core/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/memorymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"testing"
+)
+
+func TestPreStartContainer(t *testing.T) {
+	// setup vars
+
+	var tests = []struct {
+		i internalContainerLifecycleImpl
+	}{
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   nil,
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   nil,
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   nil,
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: nil,
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      nil,
+				memoryManager:   nil,
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+		{
+			i: internalContainerLifecycleImpl{
+				cpuManager:      cpumanager.NewFakeManager(),
+				memoryManager:   memorymanager.NewFakeManager(),
+				topologyManager: topologymanager.NewFakeManager(),
+			},
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
when there's a `name` field, we can then:
```
t.Run(test.name, func(t *testing.t) {
 // Test case code here
})
```

which means when tests fail we can immediately identify the failing table test